src/components/Grid/Grid.tsx

```diff
@@ -0,0 +1,725 @@
+        onFocusChange(focus.row, focus.column);
+        scrollGridTo(focus);
+      },
+      [focus, onFocusChange, scrollGridTo]
+    );
+
+    useEffect(() => {
+      clearSelectionAndFocus(true);
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [rowStart]);
+
+    const { getSelectionType, onSelection, mouseMoveCellSelect, moveSelection } =
+      useSelectionActions({
+        onCellSelect,
+        focus: focusProp ?? focus,
+        columnCount,
+        rowCount,
+        onFocusChange,
+        scrollGridTo,
+        selection: selectionProp ?? selection,
+      });
+
+    const data: ItemDataType = {
+      showRowNumber,
+      cell,
+      rowCount,
+      columnCount,
+      rounded,
+      showHeader,
+      focus,
+      getSelectionType,
+      rowHeight,
+      headerHeight,
+      rowNumberWidth,
+    };
+
+    const InnerElementType = forwardRef<HTMLDivElement, InnerElementTypeTypes>(
+      ({ children, ...containerProps }, ref) => {
+        const [minRow, maxRow, minColumn, maxColumn] = getRenderedCursor(children);
+        return (
+          <GridContainer
+            {...containerProps}
+            className={`sticky-grid__container grid-outer ${props.className ?? ""}`}
+          >
+            <GridDataContainer
+              $top={showHeader ? headerHeight : 0}
+              $left={showRowNumber ? rowNumberWidth : 0}
+              ref={ref}
+            >
+              {children}
+            </GridDataContainer>
+            {showRowNumber && (
+              <RowNumberColumn
+                scrolledHorizontal={scrolledHorizontal}
+                minRow={minRow}
+                maxRow={maxRow}
+                rowHeight={rowHeight}
+                headerHeight={headerHeight}
+                rowWidth={rowNumberWidth}
+                rowCount={rowCount}
+                rounded={rounded}
+                getSelectionType={getSelectionType}
+                showHeader={showHeader}
+                rowStart={rowStart}
+              />
+            )}
+
+            {showHeader && (
+              <Header
+                scrolledVertical={scrolledVertical}
+                showRowNumber={showRowNumber}
+                minColumn={minColumn}
+                maxColumn={maxColumn}
+                height={headerHeight}
+                columnWidth={columnWidth}
+                cell={cell}
+                rowNumberWidth={rowNumberWidth}
+                rounded={rounded}
+                getSelectionType={getSelectionType}
+                columnCount={columnCount}
+                onColumnResize={onColumnResize}
+                columnHorizontalPosition={columnHorizontalPosition}
+                setResizeCursorPosition={setResizeCursorPosition}
+              />
+            )}
+          </GridContainer>
+        );
+      }
+    );
+    useEffect(() => {
+      containerRef.current?.focus();
+    }, []);
+
+    const onMouseDown: MouseEventHandler<HTMLDivElement> = useCallback(
```

Comment:
we should be able to extract all navigation into its own hook to improve readability of this huge component.